### PR TITLE
add stronghold pay js

### PR DESCRIPTION
**Pull request by AdrienEtienne**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by orta** [comment]
This doesn't represent an npm module, can you provide us with some documentation about what it does?

**Comment by AdrienEtienne** [comment]
Yeah sure. It is the typing for our JS module.

Our Clients are importing the JavaScript directly into their html page. The is a link to the documentation: https://docs.strongholdpay.com/#stronghold-pay-js

**Review by itsseanbennett — APPROVED** [review]
Another Stronghold team member

**Comment by AdrienEtienne** [comment]
@orta is my PR provides enough documentation?

File: types/stronghold-pay-js/tslint.json [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        "space-before-function-paren": false,
+        "space-within-parens": false,
+        "strict-export-declare-modifiers": false,
+        "trim-file": false,
+        "triple-equals": false,
+        "typedef-whitespace": false,
+        "unified-signatures": false,
+        "void-return": false,
+        "whitespace": false,
+        "no-angle-bracket-type-assertion": false
```
Comment (orta) [inline types/stronghold-pay-js/tslint.json:79]: This isn't great, is there a reason why you've disabled all these rules?

**Review by orta — COMMENTED** [review]
Cool, everything looks good - the tslint config is a big red flag though. Nothing about your .d.t code seems to indicate that you'd want everything turned off?

File: types/stronghold-pay-js/tslint.json (hunk 1 above)
Comment (AdrienEtienne) [reply types/stronghold-pay-js/tslint.json]: Thanks for your comments. I run through my changes again and you are right, I didn't need all those rules.

I just made the changes in order to keep the linter happy with the minimal configuration.

**Comment by orta** [comment]
Perfect, looks good to me, thanks

**Pull request merged**